app/src/main/java/org/sagebionetworks/research/mpower/tracking/fragment/TimePickerFragment.java

```diff
@@ -0,0 +1,114 @@
+package org.sagebionetworks.research.mpower.tracking.fragment;
+
+import static com.google.common.base.Preconditions.checkState;
+
+import android.app.Dialog;
+import android.app.TimePickerDialog;
+import android.app.TimePickerDialog.OnTimeSetListener;
+import android.arch.lifecycle.ViewModelProviders;
+import android.content.Context;
+import android.os.Build;
+import android.os.Build.VERSION;
+import android.os.Bundle;
+import android.os.Parcelable;
+import android.support.annotation.NonNull;
+import android.support.v4.app.DialogFragment;
+
+import org.joda.time.Hours;
+import org.sagebionetworks.research.mpower.R;
+import org.sagebionetworks.research.mpower.tracking.model.TrackingItem;
+import org.sagebionetworks.research.mpower.tracking.model.TrackingStepView;
+import org.sagebionetworks.research.mpower.tracking.view_model.SymptomTrackingTaskViewModel;
+import org.sagebionetworks.research.mpower.tracking.view_model.TrackingTaskViewModel;
+import org.sagebionetworks.research.mpower.tracking.view_model.TrackingTaskViewModelFactory;
+import org.sagebionetworks.research.mpower.tracking.view_model.logs.SymptomLog;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.threeten.bp.DateTimeUtils;
+import org.threeten.bp.Instant;
+import org.threeten.bp.ZoneId;
+import org.threeten.bp.ZoneOffset;
+import org.threeten.bp.ZonedDateTime;
+import org.threeten.bp.temporal.ChronoUnit;
+import org.threeten.bp.temporal.TemporalUnit;
+import org.threeten.bp.zone.ZoneRulesException;
+
+import java.util.Calendar;
+
+import javax.inject.Inject;
+
+import dagger.android.support.AndroidSupportInjection;
+
+public class TimePickerFragment extends DialogFragment {
```

Comment:
Once again, this needs to be a re-usable fragment that is de-coupled from any concept of "tracked items".

I'm seeing a lot of simple data collection fragments that we spent a lot of time developing using StepLayouts, Steps, StepResults with the previous iteration of the SDK.  Is there any way we could re-use that well-tested functionality and just give it a make over to integrate with the new JSON format for colors, actions, etc?

Reply:
@liujoshua Isn't it the case the we can't have a task that is a mix of both SageResearch and ResearchSuite steps or am I mistaken?